x-pack/plugins/security_solution/common/utils/build_alert_detail_path.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { addSpaceIdToPath } from '@kbn/spaces-plugin/common';
+import { ALERTS_PATH, APP_PATH } from '../constants';
+
+export const buildAlertDetailPath = ({
+  alertId,
+  index,
+  timestamp,
+}: {
+  alertId: string;
+  index: string;
+  timestamp: string;
+}) => `${ALERTS_PATH}/redirect/${alertId}?index=${index}&timestamp=${timestamp}`;
+
+export const getAlertDetailsUrl = ({
+  alertId,
+  index,
+  timestamp,
+  basePath,
+  spaceId,
+}: {
+  alertId: string;
+  index: string;
+  timestamp: string;
+  basePath?: string;
+  spaceId?: string | null;
+}) => {
+  const alertDetailPath = buildAlertDetailPath({ alertId, index, timestamp });
+  const alertDetailPathWithAppPath = `${APP_PATH}${alertDetailPath}`;
+  return basePath
```

Comment:
`core.http.publicBaseUrl` isn't guaranteed to be set, though it's highly likely that it will be. Given that condition, we will not produce a url if the `basePath` is not set